codegen-winter/src/air/boundary_constraints.rs

```diff
@@ -48,23 +49,17 @@ pub(super) fn add_fn_get_assertions(impl_ref: &mut Impl, ir: &AirIR) {
 // ================================================================================================
 
 /// Code generation trait for generating Rust code strings from boundary constraint expressions.
-trait Codegen {
-    fn to_string(&self) -> String;
-}
-
-impl Codegen for Expr {
+impl Codegen for BoundaryExpr {
     fn to_string(&self) -> String {
         match self {
             Self::Constant(value) => format!("Felt::new({})", value),
```

Comment:
Hm I think it was left that way because our main trace assertions are Felt, but yes, it needs to be updated as part of adding codegen for aux assertions, or we need to handle them differently